sonar-db/src/test/java/org/sonar/db/webhook/WebhookDeliveryDaoTest.java

```diff
@@ -0,0 +1,131 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.webhook;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.DbTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class WebhookDeliveryDaoTest {
+
+  private static final long DATE_1 = 1_999_000L;
+  
+  @Rule
+  public final DbTester dbTester = DbTester.create(System2.INSTANCE).setDisableDefaultOrganization(true);
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private final DbClient dbClient = dbTester.getDbClient();
+  private final DbSession dbSession = dbTester.getSession();
+  private final WebhookDeliveryDao underTest = dbClient.webhookDeliveryDao();
+
+  @Test
+  public void insert_row_with_only_mandatory_columns() {
+    WebhookDeliveryDto dto = newDto("DELIVERY_1", "COMPONENT_1", "TASK_1");
+
+    underTest.insert(dbSession, dto);
+
+    WebhookDeliveryDto stored = selectByUuid(dto.getUuid());
+    verifyMandatoryFields(dto, stored);
+
+    // optional fields are null
+    assertThat(stored.getHttpStatus()).isNull();
+    assertThat(stored.getDurationMs()).isNull();
+    assertThat(stored.getErrorStacktrace()).isNull();
+  }
+
+  @Test
+  public void insert_row_with_all_columns() {
+    WebhookDeliveryDto dto = newDto("DELIVERY_1", "COMPONENT_1", "TASK_1")
+      .setDurationMs(10000)
+      .setHttpStatus(200)
+      .setErrorStacktrace("timeout")
+      .setPayload("{json}");
+
+    underTest.insert(dbSession, dto);
+
+    WebhookDeliveryDto stored = selectByUuid(dto.getUuid());
+    verifyMandatoryFields(dto, stored);
+    assertThat(stored.getHttpStatus()).isEqualTo(dto.getHttpStatus());
+    assertThat(stored.getDurationMs()).isEqualTo(dto.getDurationMs());
+    assertThat(stored.getErrorStacktrace()).isEqualTo(dto.getErrorStacktrace());
+  }
+
+  @Test
+  public void delete_rows_before_date() {
+    underTest.insert(dbSession, newDto("DELIVERY_1", "COMPONENT_1", "TASK_1").setCreatedAt(1_000_000L));
+    underTest.insert(dbSession, newDto("DELIVERY_2", "COMPONENT_1", "TASK_2").setCreatedAt(2_000_000L));
+    underTest.insert(dbSession, newDto("DELIVERY_3", "COMPONENT_2", "TASK_3").setCreatedAt(1_000_000L));
+
+    // should delete the old delivery on COMPONENT_1 and keep the one of COMPONENT_2
+    underTest.deleteComponentBeforeDate(dbSession, "COMPONENT_1", 1_500_000L);
+
+    List<Object> uuids = dbTester.select(dbSession, "select uuid as \"uuid\" from webhook_deliveries")
+      .stream()
+      .map(columns -> columns.get("uuid"))
+      .collect(Collectors.toList());
+    assertThat(uuids).containsOnly("DELIVERY_2", "DELIVERY_3");
```

Comment:
I haven't verified, but I think you could use the `extracting` method with the lamda `columns -> columns.get("uuid")`. it should work as well as the usual method reference on the DTO